### DISCUSSION TITLE
Deep Link to a mini app

### DISCUSSION THREAD

**Discussion by Wolfant** [callstack/repack#417] [Q&A]
someone maybe could implement deep link to a mini app, the deep link should redirect to the host app then authenticate and reach the mini app, can this be done?

**Comment by jbroma** [comment]
That depends on how you handle deep links inside of you app. If you do it via react-navigation it should be possible without issues. If you encounter some please create an issue so we can investigate whether this is a problem on side of Re.Pack

**Comment by sahajarora1286** [comment]
I accomplished this using the `react-navigation` package.

Make appropriate use of `LinkingConfig` in your host app, where you would define the screens for the micro-apps. In my case, all micro-app screens are unknown to the host because I want to let micro-apps manage their own screens. So what I do is that I render a generic `/app` screen and then remotely load the `EntryRouteHandler` from the remote micro-app (which it exposes via ModuleFederation config). I pass the `entryScreen` to the `EntryRouteHandler` of the remote micro-app, which has its own `NavigationContainer` and `linkingConfig`. All those things need to be synchronized properly, so there's some manual stuff that needs to be done at that level but it can be extracted out into a common component (in my case I call it `RemoteAppNavigationContainer`). The micro-app's `EntryRouteHandlers` then simply render the `RemoteAppNavigationContainer` and pass to it whatever screens (along with paths and params) they want to maintain.

**Reply by AbrahamBaby** [reply]
> I accomplished this using the `react-navigation` package.
> 
> Make appropriate use of `LinkingConfig` in your host app, where you would define the screens for the micro-apps. In my case, all micro-app screens are unknown to the host because I want to let micro-apps manage their own screens. So what I do is that I render a generic `/app` screen and then remotely load the `EntryRouteHandler` from the remote micro-app (which it exposes via ModuleFederation config). I pass the `entryScreen` to the `EntryRouteHandler` of the remote micro-app, which has its own `NavigationContainer` and `linkingConfig`. All those things need to be synchronized properly, so there's some manual stuff that needs to be done at that level but it can be extracted out into a common component (in my case I call it `RemoteAppNavigationContainer`). The micro-app's `EntryRouteHandlers` then simply render the `RemoteAppNavigationContainer` and pass to it whatever screens (along with paths and params) they want to maintain.

Hi @sahajarora1286 , 
  Great work on the deeplinking on MFE architecture, can you also share some sample code around it as i was not able to understand how the deeplink handling works along with navigation and params, if you can brief about the approach you have proposed, that would be of great help i guess
